### PR TITLE
Specify MSC3882: Using an existing session to log in another

### PR DESCRIPTION
**Pull request by turt2live**
MSC: https://github.com/matrix-org/matrix-spec-proposals/pull/3882

Preview: https://pr1530--matrix-spec-previews.netlify.app

### CODE REVIEW

File: changelogs/client_server/newsfragments/1530.new [hunk 1]
```diff
@@ -0,0 +1 @@
+[`POST /_matrix/client/v3/login/get_token`](/client-server-api/#post_matrixclientv3loginget_token).
```
Comment (richvdh) [inline changelogs/client_server/newsfragments/1530.new:1]: I wasn't aware of this type of newsfragment being a thing. We have other new endpoints (`POST /_matrix/media/v1/create` etc) which haven't had one of these...

File: data/api/client-server/login.yaml [hunk 2]
```diff
@@ -54,12 +55,22 @@ paths:
...
                         The login type. This is supplied as the `type` when
                         logging in.
                       type: string
+                    get_login_token:
+                      type: boolean
+                      x-addedInMatrixVersion: "1.7"
+                      description: |-
+                        If `type` is `m.login.token`, an optional field to indicate
+                        to the unauthenticated client that the homeserver supports
+                        the `POST /login/get_token` endpoint. Note that supporting
```
Comment (richvdh) [inline data/api/client-server/login.yaml:71]: link to endpoint pls

File: data/api/client-server/login.yaml [hunk 3]
```diff
@@ -240,3 +251,87 @@ paths:
...
+  "/login/get_token":
+    post:
+      summary: Optional endpoint to generate a single-use, time-limited, `m.login.token` token.
+      description: |-
+        Optional endpoint - the server is not required to implement this endpoint if it does not
+        intend to use or support this functionality.
+
+        This API endpoint uses the [User-Interactive Authentication API](/client-server-api/#user-interactive-authentication-api).
+
+        An already-authenticated client would call this endpoint to generate a single-use, time-limited,
```
Comment (richvdh) [inline data/api/client-server/login.yaml:263]:
```suggestion
        An already-authenticated client can call this endpoint to generate a single-use, time-limited,
```

File: data/api/client-server/login.yaml [hunk 4]
```diff
@@ -240,3 +251,87 @@ paths:
...
+      x-addedInMatrixVersion: "1.7"
+      security:
+        - accessToken: []
+      parameters:
+        - in: body
+          name: body
+          required: true
+          schema:
+            type: object
+            description: An empty JSON object.
```
Comment (richvdh) [inline data/api/client-server/login.yaml:297]: it seems like we normally include `auth` here for UIA-authenticated endpoints (eg [`/deactivate`](https://spec.matrix.org/v1.6/client-server-api/#post_matrixclientv3accountdeactivate))

**Review by richvdh — CHANGES_REQUESTED** [review]
looks good otherwise

File: changelogs/client_server/newsfragments/1530.new (hunk 1 above)
Comment (turt2live) [reply changelogs/client_server/newsfragments/1530.new:1]: Yea... They'll be caught as part of "changelog normalization" in the release process.

File: changelogs/client_server/newsfragments/1530.new [hunk 5]
```diff
@@ -1 +1 @@
-[`POST /_matrix/client/v3/login/get_token`](/client-server-api/#post_matrixclientv3loginget_token).
\ No newline at end of file
+[`POST /_matrix/client/v3/login/get_token`](/client-server-api/#post_matrixclientv1loginget_token).
```
Comment (richvdh) [inline changelogs/client_server/newsfragments/1530.new:1]:
```suggestion
[`POST /_matrix/client/v1/login/get_token`](/client-server-api/#post_matrixclientv1loginget_token).
```

File: content/client-server-api/_index.md [hunk 6]
```diff
@@ -1176,8 +1176,13 @@ client supports it, the client should redirect the user to the
 is complete, the client will need to submit a `/login` request matching
 `m.login.token`.
 
+{{< added-in v="1.7" >}} Already-authenticated clients can additionally generate
+a token for their user ID if supported by the homeserver using `POST /login/get_token`.
```
Comment (richvdh) [inline content/client-server-api/_index.md:1180]:
```suggestion
a token for their user ID if supported by the homeserver using 
[`POST /login/get_token`](/client-server-api/#post_matrixclientv1loginget_token).
```

**Review by richvdh — APPROVED** [review]
lgtm otherwise

**Pull request merged**